sql/004_search.sql

```diff
@@ -745,11 +877,16 @@ DECLARE
     exit_flag boolean := FALSE;
     batches int := 0;
     timer timestamptz := clock_timestamp();
+    pstart timestamptz;
+    pend timestamptz;
+    pcurs refcursor;
+    search_where search_wheres%ROWTYPE;
 BEGIN
 searches := search_query(_search);
 _where := searches._where;
 orderby := searches.orderby;
-total_count := coalesce(searches.total_count, searches.estimated_count);
+search_where := where_stats(_where);
+total_count := coalesce(search_where.total_count, search_where.estimated_count);
```

Comment:
Is total_count calculated even if context == 'off'? Or is this lazily evaluated and doesn't perform it's operation until the jsonb_build_object call below?

Reply:
total_count and estimated_count are both calculated in the where_stats function. estimated_count is always calculated (it's a free side effect of what we use to get the list of partitions) and total_count is not calculated in there if the context is set to 'off'. The coalesce here takes the first non-null argument, so it will use the total if it was either calculated or existed already in the cache table.